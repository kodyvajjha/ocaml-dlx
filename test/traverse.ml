let () =
  let items = [ "a"; "b"; "c"; "d"; "e"; "f"; "g" ] in
  let options =
    [
      [ "c"; "e" ];
      [ "a"; "d"; "g" ];
      [ "b"; "c"; "f" ];
      [ "a"; "d"; "f" ];
      [ "b"; "g" ];
      [ "d"; "e"; "g" ];
    ]
  in
  let cur = Dlx.Algox.mk ~items ~options in

  let node_of = Dlx.Algox.find_header_node ~name:"c" ~items cur.root in
  CCFormat.printf "@.Name: %a, id: %d@."
    CCFormat.(some string)
    node_of.name node_of.id;
  CCFormat.printf "@.%a" Dlx.Algox.pp_node cur.nodes.(10)
