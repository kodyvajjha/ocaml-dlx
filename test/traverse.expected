
Name: c, id: 3

┌───────────┐
│ 10        │
├─────┬┬────┤
│TOP:5││LEN:│
├─┬──┬┴┼────┤
│U│D │L│R   │
├─┼──┼─┼────┤
│5│28│ │    │
└─┴──┴─┴────┘